ice/trace.py

```diff
@@ -194,3 +261,56 @@ def __new__(mcls, name, bases, namespace):
 
 class TracedABC(metaclass=TracedABCMeta):
     ...
+
+
+def get_first_descendant(value):
+    if value:
+        if isinstance(value, dict):
+            first, *_ = value.values()
+            return get_first_descendant(first)
+        if isinstance(value, (list, tuple)):
+            if isinstance(value[0], str):
+                return [v for v in value if isinstance(v, str)]
+            return get_first_descendant(value[0])
+    return value
+
+
+def get_strings(value) -> list[str]:
+    """
+    Represent the given value as a short list of short strings
+    that can be stored directly in the central trace file and loaded eagerly in the UI.
+    """
+    if isinstance(value, dict) and "value" in value:
+        value = value["value"]
+
+    if isinstance(value, dict):
+        value = {k: v for k, v in value.items() if k not in ("self", "record")}
+
+    result = get_first_descendant(value)
+
+    if isinstance(result, tuple):
+        result = list(result)
+    if not (result and isinstance(result, list)):
+        if result in (None, (), "", [], {}):
+            result = "()"
+        result = [str(result)]
+
+    result = get_short_list(result)
+    result = [get_short_string(v) for v in result]
+    return result
+
+
+def get_short_string(string, max_length=35) -> str:
+    return string[:max_length].strip() + "..." if len(string) > max_length else string
+
+
+def get_short_list(lst: list, max_length=3) -> list:
+    return lst[:max_length] + ["..."] if len(lst) > max_length else lst
```

Comment:
`get_strings` and its support code is kind of messy and hard to follow. (I did a sloppy job of code-reviewing the TracePage versions.) Can you add a TODO to simplify and restructure them?